src/Proto.Actor/Future/Futures.cs

```diff
@@ -51,10 +47,7 @@ await using (cancellationToken.Register(() => _tcs.TrySetCanceled()))
             }
             catch
             {
-                if (!_system.Metrics.IsNoop)
-                {
-                    _metrics!.FuturesTimedOutCount.Inc(new[] {System.Id, _system.Address});
-                }
+                _metrics?.FuturesTimedOutCount.Inc(new[] {System.Id, System.Address});
```

Comment:
is metrics changed to be null instead of Noop provider as default?
If not, we don't want to touch the metrics API if the noop provider is present.
as the Inc allocates a new object array each call

Reply:
I think metrics actually warrants some redesign overall. The abstraction here should not pass any system level parameters at all, but expose a parameterless .Inc() method.

This will simplify usage, and internally we can pre-assign the system parameters and not do it every call..

Reply:
And No-ops could be called without checking if it actually is a no-op.

Reply:
There are plenty of other cases where that is relevant though. e.g. message type, actor type etc.
We still need a pattern that is intact for this. in the current form, the Noop is what we have

Reply:
You would still pass the specific metric values at the call point, I was just thinking we did not need to pass the values that are common for the entire system. It would have made the interfaces a little cleaner IMO